src/main/java/de/chojo/repbot/dao/pagination/PageAccess.java

```diff
@@ -31,4 +34,23 @@ public int pages() {
     public List<T> page(int page) {
         return pageSupplier.apply(page);
     }
+
+    @NotNull
+    @Override
+    public Iterator<List<T>> iterator() {
+        return new Iterator<>() {
+            private int currPage = 0;
+            private final int pages = pages();
+
+            @Override
+            public boolean hasNext() {
+                return currPage < pages;
+            }
+
+            @Override
+            public List<T> next() {
+                return page(currPage++);
+            }
+        };
+    }
```

Comment:
Added iterator to pages. This makes it easier to read the paged results if it is used for pages iteration, which we are using in two cases already